src/jsp/com/cubrid/plcsql/compiler/ParseTreeConverter.java

```diff
@@ -0,0 +1,1912 @@
+        }
+
+        if (targetType == null) {
+            return (Expr) visit(ctx);
+        } else {
+            Expr e = (Expr) visit(ctx);
+            if (e instanceof ExprCast) {
+                ((ExprCast) e).setTargetType(targetType);
+            }
+            return e;
+        }
+    }
+
+    private static final Map<String, String> pcsToJavaTypeMap = new TreeMap<>();
+
+    static {
+        pcsToJavaTypeMap.put("BOOLEAN", "java.lang.Boolean");
+
+        pcsToJavaTypeMap.put("CHAR", "java.lang.String");
+        pcsToJavaTypeMap.put("VARCHAR", "java.lang.String");
+        pcsToJavaTypeMap.put("STRING", "java.lang.String");
+
+        pcsToJavaTypeMap.put("NUMERIC", "java.math.BigDecimal");
+        pcsToJavaTypeMap.put("DECIMAL", "java.math.BigDecimal");
+        pcsToJavaTypeMap.put("SHORT", "java.lang.Short");
+        pcsToJavaTypeMap.put("SMALLINT", "java.lang.Short");
+        pcsToJavaTypeMap.put("INT", "java.lang.Integer");
+        pcsToJavaTypeMap.put("INTEGER", "java.lang.Integer");
+        pcsToJavaTypeMap.put("BIGINT", "java.lang.Long");
+
+        pcsToJavaTypeMap.put("FLOAT", "java.lang.Float");
+        pcsToJavaTypeMap.put("REAL", "java.lang.Float");
+        pcsToJavaTypeMap.put("DOUBLE", "java.lang.Double");
+        pcsToJavaTypeMap.put("DOUBLE PRECISION", "java.lang.Double");
+
+        pcsToJavaTypeMap.put("DATE", "java.time.LocalDate");
+        pcsToJavaTypeMap.put("TIME", "java.time.LocalTime");
+        pcsToJavaTypeMap.put("TIMESTAMP", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIME", "java.time.LocalDateTime");
+        pcsToJavaTypeMap.put("TIMESTAMPTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("TIMESTAMPLTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIMETZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("DATETIMELTZ", "java.time.ZonedDateTime");
+        pcsToJavaTypeMap.put("SET", "java.util.Set");
+        pcsToJavaTypeMap.put("MULTISET", "org.apache.commons.collections4.MultiSet");
+        pcsToJavaTypeMap.put("LIST", "java.util.List");
+        pcsToJavaTypeMap.put("SEQUENCE", "java.util.List");
+        pcsToJavaTypeMap.put("SYS_REFCURSOR", "Query");
+    }
+
+    private static boolean isAssignableTo(Expr expr) {
+
+        if (expr instanceof ExprId) {
+            ExprId id = (ExprId) expr;
+            if (id.decl instanceof DeclVar || id.decl instanceof DeclParamOut) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static String quotedStrToJavaStr(String val) {
+        val = val.substring(1, val.length() - 1); // strip enclosing '
+        val = val.replace("''", "'");
+        return StringEscapeUtils.escapeJava(val);
+    }
+
+    private static final DateFormat dbgFormat =
+            new SimpleDateFormat("G yyyy-MM-dd HH:mm:ss.SSS XXX", Locale.US);
+
+    static {
+        dbgFormat.setTimeZone(TimeZone.getTimeZone("GMT+0"));
+    }
```

Comment:
It seems not used anywhere. maybe copied into DateTimeParsr.java?

Reply:
It was used for debugging code which is now deleted.
Let me comment out these lines.